packages/ethers/src/client.ts

```diff
@@ -587,28 +588,22 @@ export class Web3Modal extends Web3ModalScaffold {
     window?.localStorage.setItem(EthersConstantsUtil.WALLET_ID, ConstantsUtil.EMAIL_CONNECTOR_ID)
 
     if (this.emailProvider) {
-      const preferredAccountType = W3mFrameHelpers.getPreferredAccountType()
-      const [{ address, chainId, smartAccountDeployed }, { smartAccountEnabledNetworks }] =
-        await Promise.all([
-          this.emailProvider.connect({
-            chainId: this.getChainId(),
-            preferredAccountType
-          }),
-          this.emailProvider.getSmartAccountEnabledNetworks()
-        ])
-      super.setLoading(false)
+      const { address, chainId, smartAccountDeployed, preferredAccountType } =
+        await this.emailProvider.connect({ chainId: this.getChainId() })
+
       if (address && chainId) {
         EthersStoreUtil.setChainId(chainId)
         EthersStoreUtil.setProviderType(ConstantsUtil.EMAIL_CONNECTOR_ID as 'w3mEmail')
         EthersStoreUtil.setProvider(this.emailProvider as unknown as CombinedProvider)
         EthersStoreUtil.setIsConnected(true)
         EthersStoreUtil.setAddress(address as Address)
+        EthersStoreUtil.setPreferredAccountType(preferredAccountType as W3mFrameTypes.AccountType)
         this.setSmartAccountDeployed(Boolean(smartAccountDeployed))
-        this.setSmartAccountEnabledNetworks(smartAccountEnabledNetworks)
 
         this.watchEmail()
         this.watchModal()
       }
+      super.setLoading(false)
```

Comment:
Seems like a fix

Reply:
more like an improvement I think, makes it less jumpy